Client/Frontend/Sync/BraveCore/BraveCoreDataImportExportUtility.swift

```diff
@@ -0,0 +1,475 @@
+        try rethrow(self.writeNodes(Bookmark.getAllTopLevelBookmarks(context), to: fileHandle), error: .errorWritingNode)
+        decrementIndent()
+        
+        //End Header
+        try rethrow(fileHandle.write(FolderWriter.kFolderChildrenEnd), error: .errorWritingNode)
+        
+        //Flush To Disk
+        fileHandle.synchronizeFile()
+    }
+    
+    private func writeNodes(_ bookmarks: [Bookmark], to file: FileHandle) throws {
+        for bookmark in bookmarks.sorted(by: { $0.isFolder && !$1.isFolder }) {
+            if bookmark.isFolder {
+                try file.write(currentIndentation)
+                
+                //Start Folder
+                try file.write(FolderWriter.kFolderStart)
+                
+                //Created Date
+                try file.write("\(Int(bookmark.created?.timeIntervalSince1970 ?? Date().timeIntervalSince1970))")
+                
+                //Last Modified Date
+                try file.write(FolderWriter.kLastModified)
+                try file.write("\(Int(bookmark.lastVisited?.timeIntervalSince1970 ?? Date().timeIntervalSince1970))")
+                
+                //End Attributes
+                try file.write(FolderWriter.kFolderAttributeEnd)
+                
+                //Title
+                try file.write((bookmark.customTitle ?? "(No Title)").escapeForHTML())
+                
+                //End Folder
+                try file.write(FolderWriter.kFolderEnd)
+                try file.write(BraveCoreDataImportExportUtility.kNewline)
+                
+                //Children
+                if let children = bookmark.children, !children.isEmpty {
+                    try file.write(currentIndentation)
+                    try file.write(FolderWriter.kFolderChildren)
+                    try file.write(BraveCoreDataImportExportUtility.kNewline)
+                    incrementIndent()
+                    
+                    //Recursive Children
+                    try self.writeNodes(Array(children), to: file)
+                    
+                    decrementIndent()
+                    try file.write(currentIndentation)
+                    try file.write(FolderWriter.kFolderChildrenEnd)
+                    try file.write(BraveCoreDataImportExportUtility.kNewline)
+                }
+            } else {
+                try file.write(currentIndentation)
+                
+                //Start Bookmark
+                try file.write(BookmarkWriter.kBookmarkStart)
+                
+                //URL
+                try file.write(bookmark.url?.escapeAsHTMLQuotes() ?? "")
+                
+                //Created Date
+                try file.write("\(Int(bookmark.created?.timeIntervalSince1970 ?? Date().timeIntervalSince1970))")
+                
+                //ICON (OPTIONAL) - 16x16 PNG
+                if let iconUrl = bookmark.domain?.favicon?.url,
+                   let url = URL(string: iconUrl),
```

Comment:
if i understand correctly, this fetches all the bookmark favicons on import. i think this is somewhat of a privacy issue for first-time syncs because anyone on the network can see what domains you've bookmarked (because these requests occur so close together). @darkdh do you know if chromium has this issue?

after a favicon is written to disk, do we always display it from disk?

Reply:
Chromium usually get it from `ICON` attribute which is basically a data url and will be used by 
https://source.chromium.org/chromium/chromium/src/+/master:chrome/utility/importer/bookmark_html_reader.cc;l=221-224;drc=929e790ad17a54008b3dac86cbe30751c968c2ce

Reply:
@diracdeltas this file isn't used anymore. Nothing in this file is used. It will be completely removed. We only use `BraveCoreImportExportUtility.swift`, not `BraveCoreDataImportExportUtility.swift`. Sorry for this confusion :( I forgot to remove it.

Reply:
ah thanks for clarifying